### PR TITLE
Docs: Update release notes to include the Helm changes

### PR DESCRIPTION
**Pull request by 09jvilla**
Takes the draft release notes and adds all the helm improvements on top. 

Original content for the helm changes can be found here:
https://docs.google.com/document/d/1h4r3Nr5nqijUrGDUg0ODwENG3Igdp2db6R8jbZdjHSg/edit

### CODE REVIEW

File: docs/sources/release-notes/v2.2.md [hunk 1]
```diff
@@ -25,14 +25,27 @@ For the complete list of changes, see the [Changelog](https://github.com/grafana
...
+  - We’ve significantly leveled up the configuration experience for the Helm chart. A few of the most salient changes:
+    - We’ve added an `extraEnvFrom` capability to all Mimir services to enable injecting secrets via environment variables.
+    - We’ve made it possible to globally set environment variables and inject secrets across all pods in the chart using `global.extraEnv` and `global.extraEnvFrom`. Note that the memcached and minio pods are not included.
+    - We’ve switched the default storage of the Mimir configuration from a `Secret` to a `ConfigMap`, which makes it easier to quickly diff your Mimir configuration between upgrades. We especially like the [Helm diff](https://github.com/databus23/helm-diff) plugin for this purpose.
+    - We’ve added a structuredConfig option which allows overwriting specific key<>value pairs in the `mimir.config` template, saving you from having to maintain the entire `mimir.config` in your own values.yaml.
```
Comment (pracucci) [inline docs/sources/release-notes/v2.2.md:42]:
```suggestion
    - We’ve added a `structuredConfig` option which allows overwriting specific key<>value pairs in the `mimir.config` template, saving you from having to maintain the entire `mimir.config` in your own `values.yaml`.
```

File: docs/sources/release-notes/v2.2.md [hunk 2]
```diff
@@ -25,14 +25,27 @@ For the complete list of changes, see the [Changelog](https://github.com/grafana
...
+    - We’ve added the ability to create global pod annotations. This unlocks the ability to trigger a restart of all services in response to a single event, like the update of the secret containing Mimir’s storage credentials. 
+  - We’ve set the chart to disable `ingester.ring.unregister-on-shutdown` and `distributor.extend-writes` to make for a smoother upgrade experience. Now rolling restarts of ingesters should be less likely to cause spikes in resource usage.
```
Comment (pracucci) [inline docs/sources/release-notes/v2.2.md:44]:
```suggestion
  - We’ve set the chart to disable `-ingester.ring.unregister-on-shutdown` and `-distributor.extend-writes` to make for a smoother upgrade experience. Now rolling restarts of ingesters should be less likely to cause spikes in resource usage. 
```

**Review by pracucci — COMMENTED** [review]
Nice! Few nits.

**Comment by osg-grafana** [comment]
Change any instance of "allows -ing" to "allows you to <VERB>".

File: docs/sources/release-notes/v2.2.md [hunk 3]
```diff
@@ -25,14 +25,27 @@ For the complete list of changes, see the [Changelog](https://github.com/grafana
...
   Feature can be enabled with new experimental flag `-blocks-storage.bucket-store.index-header.map-populate-enabled`.
 
 - **Faster ingester startup**: Ingesters now replay Write-Ahead-Log by about 50% faster, and they also re-join the ring sooner under some conditions.
 
+- **Helm Chart improvements**: The [Mimir Helm chart](https://github.com/grafana/mimir/tree/main/operations/helm/charts/mimir-distributed) is our recommended way to install Mimir on Kubernetes. As part of the Mimir 2.2 release, we're also releasing a major version bump to the Helm chart. Below we mention some of the most notable enhancements. For the full list, see the [Helm chart changelog](FIXME LINK).
+  - The Helm chart now supports installing Mimir on [OpenShift](https://www.redhat.com/en/technologies/cloud-computing/openshift). 
+  - The Helm chart can now easily deploy the [Grafana Agent](https://github.com/grafana/agent) to scrape metrics and logs from all Mimir pods and ship them to a remote store, making it easy to monitor the health of your Mimir installation. For more information on how to set this up, see [FIXME](fixme - link to docs on this).
```
Comment (colega) [inline docs/sources/release-notes/v2.2.md:36]: I understand that all these FIXMEs are blockers for this PR, right?

File: docs/sources/release-notes/v2.2.md [hunk 4]
```diff
@@ -25,14 +25,27 @@ For the complete list of changes, see the [Changelog](https://github.com/grafana
...
 
-- **Helm Chart update**: TBD
-
 - **Store-gateway can now optionally prepopulate the file system cache when memory-mapping index-header files.**
   This can help store-gateway to avoid looking stuck while loading index-headers.
   Feature can be enabled with new experimental flag `-blocks-storage.bucket-store.index-header.map-populate-enabled`.
 
 - **Faster ingester startup**: Ingesters now replay Write-Ahead-Log by about 50% faster, and they also re-join the ring sooner under some conditions.
 
+- **Helm Chart improvements**: The [Mimir Helm chart](https://github.com/grafana/mimir/tree/main/operations/helm/charts/mimir-distributed) is our recommended way to install Mimir on Kubernetes. As part of the Mimir 2.2 release, we're also releasing a major version bump to the Helm chart. Below we mention some of the most notable enhancements. For the full list, see the [Helm chart changelog](FIXME LINK).
```
[the remark this answers is not in the document]
Comment (dimitarvdimitrov) [reply docs/sources/release-notes/v2.2.md]: done in d68fe710aba16f6b55423e7b806ba35e9f292501

File: docs/sources/release-notes/v2.2.md (hunk 3 above)
Comment (dimitarvdimitrov) [reply docs/sources/release-notes/v2.2.md]: they are and I think I replaced all of them in https://github.com/grafana/mimir/pull/2301/commits/210dca4a7fb100ccfcc21e8c2b2f17e4ae6749d6 . In https://github.com/grafana/mimir/pull/2301/commits/c68d880d5a1591fba9859039b4014eeb5335b9d4 I also removed one which talked about the smoke-test tool, which wasn't included in the getting started guide

File: docs/sources/release-notes/v2.2.md [hunk 5]
```diff
@@ -25,15 +25,27 @@ This feature still needs additional testing; we do not recommend using it in pro
...
-- **Helm Chart update**: TBD
-
 - **Store-gateway performance optimization**
   The store-gateway can now pre-populate the file system cache when memory-mapping index-header files.
   This can help the store-gateway avoid looking stuck while loading [index-headers]({{< relref "../operators-guide/architecture/binary-index-header" >}}). 
   This feature is experimental and disabled by default; enable it using the flag `-blocks-storage.bucket-store.index-header.map-populate-enabled`.
 
 - **Faster ingester startup**: Ingesters now replay their Write-Ahead-Logs about 50% faster, and they also re-join the ring sooner under some conditions.
 
+- **Helm Chart improvements**: The [Mimir Helm chart](https://github.com/grafana/mimir/tree/main/operations/helm/charts/mimir-distributed) is our recommended way to install Mimir on Kubernetes. As part of the Mimir 2.2 release, we're also releasing a version 3.0 of the Helm chart. Below we mention some of the most notable enhancements. For the full list, see the [Helm chart changelog](https://github.com/grafana/mimir/tree/main/operations/helm/charts/mimir-distributed/CHANGELOG.md).
```
Comment (osg-grafana) [inline docs/sources/release-notes/v2.2.md:35]:
```suggestion
- **Helm Chart improvements**: The [Mimir Helm chart](https://github.com/grafana/mimir/tree/main/operations/helm/charts/mimir-distributed) is the best way to install Mimir on Kubernetes. As part of the Mimir 2.2 release, we're also releasing version 3.0 of the Helm chart. Notable enhancements follow. For the full list of changes, see the [Helm chart changelog](https://github.com/grafana/mimir/tree/main/operations/helm/charts/mimir-distributed/CHANGELOG.md).
```

File: docs/sources/release-notes/v2.2.md [hunk 6]
```diff
@@ -25,15 +25,27 @@ This feature still needs additional testing; we do not recommend using it in pro
...
+  - The Helm chart now supports [OpenShift](https://www.redhat.com/en/technologies/cloud-computing/openshift). 
+  - The Helm chart can now easily deploy the [Grafana Agent](https://github.com/grafana/agent) to scrape metrics and logs from all Mimir pods and ship them to a remote store, making it easy to monitor the health of your Mimir installation. For more information on how to set this up, see [Collecting metrics and logs from Grafana Mimir]({{< relref "../operators-guide/monitoring-grafana-mimir/collecting-metrics-and-logs.md" >}}).
+  - The Helm chart now enables multi-tenancy by default. This makes it easy for users to add tenants as they grow their cluster and take advantage of Mimir's per-tenant quality-of-service features, which improve its stability and resilience at high scale. To learn more about how multi-tenancy in Mimir works, see [Grafana Mimir authorization and authentication]({{< relref "../operators-guide/securing/authentication-and-authorization" >}}). We've implemented this change in a way that is backwards compatible. To read about how we did this, see [#2117](https://github.com/grafana/mimir/pull/2117).
```
Comment (osg-grafana) [inline docs/sources/release-notes/v2.2.md:38]:
```suggestion
  - The Helm chart now enables multi-tenancy by default. This makes it easy for you to add tenants as you grow your cluster. You can take advantage of Mimir's per-tenant quality-of-service features, which improves stability and resilience at high scale. To learn more about how multi-tenancy in Mimir works, see [Grafana Mimir authorization and authentication]({{< relref "../operators-guide/securing/authentication-and-authorization" >}}). This change is backwards-compatible. To read about how we implemented this, see [#2117](https://github.com/grafana/mimir/pull/2117). 
```

File: docs/sources/release-notes/v2.2.md [hunk 7]
```diff
@@ -25,15 +25,27 @@ This feature still needs additional testing; we do not recommend using it in pro
...
+  - The Helm chart can now easily deploy the [Grafana Agent](https://github.com/grafana/agent) to scrape metrics and logs from all Mimir pods and ship them to a remote store, making it easy to monitor the health of your Mimir installation. For more information on how to set this up, see [Collecting metrics and logs from Grafana Mimir]({{< relref "../operators-guide/monitoring-grafana-mimir/collecting-metrics-and-logs.md" >}}).
+  - The Helm chart now enables multi-tenancy by default. This makes it easy for users to add tenants as they grow their cluster and take advantage of Mimir's per-tenant quality-of-service features, which improve its stability and resilience at high scale. To learn more about how multi-tenancy in Mimir works, see [Grafana Mimir authorization and authentication]({{< relref "../operators-guide/securing/authentication-and-authorization" >}}). We've implemented this change in a way that is backwards compatible. To read about how we did this, see [#2117](https://github.com/grafana/mimir/pull/2117). 
+  - We've significantly leveled up the configuration experience for the Helm chart. A few of the most salient changes:
```
Comment (osg-grafana) [inline docs/sources/release-notes/v2.2.md:39]:
```suggestion
  - We have significantly improved the configuration experience for the Helm chart, and here are a few of the most salient changes:
```

File: docs/sources/release-notes/v2.2.md [hunk 8]
```diff
@@ -25,15 +25,27 @@ This feature still needs additional testing; we do not recommend using it in pro
...
+    - We've added an `extraEnvFrom` capability to all Mimir services to enable you to inject secrets via environment variables.
+    - We've made it possible to globally set environment variables and inject secrets across all pods in the chart using `global.extraEnv` and `global.extraEnvFrom`. Note that the memcached and minio pods are not included.
+    - We've switched the default storage of the Mimir configuration from a `Secret` to a `ConfigMap`, which makes it easier to quickly diff your Mimir configuration between upgrades. We especially like the [Helm diff](https://github.com/databus23/helm-diff) plugin for this purpose.
```
Comment (osg-grafana) [inline docs/sources/release-notes/v2.2.md:42]:
```suggestion
    - We've switched the default storage of the Mimir configuration from a `Secret` to a `ConfigMap`, which makes it easier to quickly see the differences between your Mimir configurations between upgrades. We especially like the [Helm diff](https://github.com/databus23/helm-diff) plugin for this purpose.
```

File: docs/sources/release-notes/v2.2.md [hunk 9]
```diff
@@ -25,15 +25,27 @@ This feature still needs additional testing; we do not recommend using it in pro
...
+    - We've added a `structuredConfig` option which allows you to overwrite specific key<>value pairs in the `mimir.config` template, saving you from having to maintain the entire `mimir.config` in your own `values.yaml`.
```
Comment (osg-grafana) [inline docs/sources/release-notes/v2.2.md:43]:
```suggestion
    - We've added a `structuredConfig` option, which allows you to overwrite specific key<>value pairs in the `mimir.config` template, saving you from having to maintain the entire `mimir.config` in your own `values.yaml`.
```
What is "key<>value". Can we simply write "key-value" in plain English? If not, use `key<>value` instead.

File: docs/sources/release-notes/v2.2.md [hunk 10]
```diff
@@ -25,15 +25,27 @@ This feature still needs additional testing; we do not recommend using it in pro
...
+    - We've added the ability to create global pod annotations. This unlocks the ability to trigger a restart of all services in response to a single event, like the update of the secret containing Mimir's storage credentials.
```
Comment (osg-grafana) [inline docs/sources/release-notes/v2.2.md:44]:
```suggestion
    - We've added the ability to create global pod annotations. This unlocks the ability to trigger a restart of all services in response to a single event, like the update of the secret containing Mimir's storage credentials. 
```
For clarity, which punctuation is accurate:
* global-pod annotations
* global, pod annotations

File: docs/sources/release-notes/v2.2.md [hunk 11]
```diff
@@ -25,15 +25,27 @@ This feature still needs additional testing; we do not recommend using it in pro
...
+    - We've switched the default storage of the Mimir configuration from a `Secret` to a `ConfigMap`, which makes it easier to quickly diff your Mimir configuration between upgrades. We especially like the [Helm diff](https://github.com/databus23/helm-diff) plugin for this purpose.
+    - We've added a `structuredConfig` option which allows you to overwrite specific key<>value pairs in the `mimir.config` template, saving you from having to maintain the entire `mimir.config` in your own `values.yaml`.
+    - We've added the ability to create global pod annotations. This unlocks the ability to trigger a restart of all services in response to a single event, like the update of the secret containing Mimir's storage credentials. 
+  - We've set the chart to disable `-ingester.ring.unregister-on-shutdown` and `-distributor.extend-writes` to make for a smoother upgrade experience. Now rolling restarts of ingesters should be less likely to cause spikes in resource usage.
```
Comment (osg-grafana) [inline docs/sources/release-notes/v2.2.md:45]:
```suggestion
  - We've set the chart to disable `-ingester.ring.unregister-on-shutdown` and `-distributor.extend-writes`, for a smoother upgrade experience. Rolling restarts of ingesters are now less likely to cause spikes in resource usage. 
```

File: docs/sources/release-notes/v2.2.md [hunk 12]
```diff
@@ -25,15 +25,27 @@ This feature still needs additional testing; we do not recommend using it in pro
...
+    - We've added a `structuredConfig` option which allows you to overwrite specific key<>value pairs in the `mimir.config` template, saving you from having to maintain the entire `mimir.config` in your own `values.yaml`.
+    - We've added the ability to create global pod annotations. This unlocks the ability to trigger a restart of all services in response to a single event, like the update of the secret containing Mimir's storage credentials. 
+  - We've set the chart to disable `-ingester.ring.unregister-on-shutdown` and `-distributor.extend-writes` to make for a smoother upgrade experience. Now rolling restarts of ingesters should be less likely to cause spikes in resource usage. 
+  - We've improved the documentation experience for the Helm chart by adding a [Getting started with Mimir using the Helm chart]({{< relref "../operators-guide/deploy-grafana-mimir/getting-started-helm-charts.md" >}}). 
+  - We've added a smoke test for your Mimir cluster to help catch errors immediately after you install or upgrade via Helm.
```
Comment (osg-grafana) [inline docs/sources/release-notes/v2.2.md:47]:
```suggestion
  - We've added a smoke test for your Mimir cluster to help catch errors immediately after you install or upgrade Mimir via the Helm chart.
```
Vet this for technical accuracy.

**Review by osg-grafana — APPROVED** [review]
Unblocking with feedback
